test/helper.js

```diff
@@ -3,58 +3,42 @@ var data        = require('../hooks/data');
 var taskcluster = require('taskcluster-client');
 var taskcreator = require('../hooks/taskcreator');
 var v1          = require('../routes/v1');
-var bin = {
-  server:             require('../bin/server')
-};
+var load        = require('../bin/main');
+var config      = require('typed-env-config');
+var _           = require('lodash');
 
-var testProfile = 'test';
+var loadOptions = {profile: 'test', process: 'test-helper'};
+var cfg = config({profile: 'test'});
```

Comment:
I can't use `load('cfg', loadOptions)` because it returns a promise...

I suggest that we do what I've done in other modules, make a single global `mocha.before()` hook... So we don't restart the server between suites.. We can still call `await helper.Hook.scan({}, {handler: hook => hook.remove()});`  before each test case... That way all persisted state in the api server is reset...


--------
Also I see how MockTaskCreator is nice.. but honestly for this module I wouldn't have done anything by integration tests, as it's really the only thing we care about...
Hence I would give the tasks a special route `dummy.hooks-test.<slugid>` and just listen on pulse to see them be created... 
`base.testing.PulseTestReceiver` is good for this...
See: https://github.com/taskcluster/taskcluster-base/blob/master/testing.js#L32-L49

I normally configure `base.testing.PulseTestReceiver` in the helper and expose it on the helper module.
I know this is ugly, but is it really worth the effort to write anything more than integration tests when our projects are this simple...?

I like the integration tests because they mimick production the closest, and they remain stable as long as the API is stable. Anyways, that why I don't mock things and write unit-tests, but more like black box integration tests...

I see how auth needed the unit tests especially with the dfa stuff... I also see how aws-provisioner needs to mock things because cleanup is expensive.

-----
Anyways, this doesn't need azure creds anymore... just tc creds with scopes for the `jungle` azure storage account :)
The jungle account is intended for abuse...